ui/src/app/customizations/tech-extensions/detail/tech-extension-detail.component.ts

```diff
@@ -1,14 +1,32 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+import { Subscription } from 'rxjs/Subscription';
+import { Extension } from '../../../model';
 import { ExtensionStore } from '../../../store/extension/extension.store';
 
 @Component({
   selector: 'syndesis-tech-extension-detail',
   templateUrl: 'tech-extension-detail.component.html'
 })
-export class TechExtensionDetailComponent implements OnInit {
-  constructor(private store: ExtensionStore) { }
+export class TechExtensionDetailComponent implements OnInit, OnDestroy {
+  extension: Extension;
+  loading: Observable<boolean>;
+  subscription: Subscription;
+  constructor(private store: ExtensionStore,
+              private router: Router,
+              private route: ActivatedRoute) {
+    this.loading = store.loading;
+               }
 
   ngOnInit() {
-    // WIP
+    this.subscription = this.store.resource.subscribe(extension => {
+      this.extension = extension;
+    });
```

Comment:
You can streamline the component implementation by using observables as is, so you can declutter the component by removing subscriptions and stuff. First, turn the `extension`class field into this:

`extension$: Observable<Extension>;`

And then remove the subscription above and use this expression instead:

`this.extension$ = this.store.resource;`

TIP: If you name object instances by the camelCased version of its injected type - `ExtensionStore` in this case, code becomes more expressive. The above would look more self-explanatory if it was 

`this.extension$ = this.extensionStore.resource;`

Now you can completely remove the `subscription: Subscription` field and the `ngOnDestroy` blocks below. The only action required then will be to go the components template and apply this simple change:

```html
<syndesis-loading [loading]="loading | async">
   <div class="row" *ngIf="extension$ | async; let extension = extension$">
```

And that's it! ;)

Reply:
FYI, the Angular engine automatically "completes" any observable binding piped with the `async` pipe. That is why it is recommended to bind observable information directly, rather than subscribing to it (which entails tearing down the subscriptions by hand upon disposing the component and stuff)

Reply:
Will do, thanks!